ocean/nemo/nemolite2d/manual_versions/psykal_dm/time_step_mod.f90

```diff
@@ -0,0 +1,177 @@
+module time_step_mod
+  implicit none
+
+  private
+
+  public invoke_time_step
+
+contains
+
+  subroutine invoke_time_step(istp, ssha, ssha_u, ssha_v, &
+                              sshn_t, sshn_u, sshn_v, &
+                              hu, hv, ht, ua, va, un, vn)
+    use kind_params_mod
+    use dl_timer
+    use field_mod
+    use grid_mod
+    use model_mod,       only: rdt, cbfr, visc
+    use physical_params_mod, only: g, omega, d2r
+    use momentum_mod,    only: momentum_v_code
+    use momentum_mod,    only: momentum_u_code
+    use continuity_mod,  only: continuity_code
+    use time_update_mod, only: next_sshu_code, next_sshv_code
+    use boundary_conditions_mod
+    implicit none
+    integer,         intent(in)    :: istp
+    type(r2d_field), intent(inout) :: un, vn, sshn_t, sshn_u, sshn_v
+    type(r2d_field), intent(inout) :: ua, va, ssha, ssha_u, ssha_v
+    type(r2d_field), intent(in)    :: hu, hv, ht
+    ! Locals
+    integer :: it, ji, jj
+    integer :: idepth = 1
+
+    ! We don't know the state of a field's halos on entry to this
+    ! invoke so we have to check...
+    ! (hu, hv and ht are constant so no need to worry about them)
+!!$    if(sshn_u%is_dirty(depth=idepth))then
```

Comment:
Are this checks not needed anymore? In this case remove and update the comment.

Reply:
They aren't strictly needed in this manual version and dl_esm_inf doesn't yet have the necessary support. I've removed the commented-out code and updated the comment with a reference to the Issue in dl_esm_inf.